src/client/cursor.rs

```diff
@@ -14,18 +18,16 @@ use std::io::{Read, Write};
 ///                          beginning with the database name and a period.
 /// `batch_size` - How many documents to fetch at a given time from the server.
 /// `cursor_id` - Uniquely identifies the cursor being returned by the reply.
-pub struct Cursor<'a, T> where T: Read + Write + 'a {
-    request_id: i32,
-    namespace: String,
+pub struct Cursor<'a> {
+    collection: &'a Collection<'a>,
     batch_size: i32,
     cursor_id: i64,
     limit: i32,
     count: i32,
     buffer: VecDeque<bson::Document>,
-    stream: &'a mut T,
 }
 
-impl <'a, T> Cursor<'a, T> where T: Read + Write + 'a {
+impl <'a> Cursor<'a> {
```

Comment:
There is a `next()` method (implemented by @saghmrossi for Iterator). I'll add `has_next()`.